modules/schema/src/main/java/org/apache/ignite/internal/schema/Column.java

```diff
@@ -205,8 +205,12 @@ public int hashCode() {
      * @param val Object to validate.
      */
     public void validate(@Nullable Object val) {
-        if (val == null && !nullable) {
-            throw new SchemaMismatchException(nullConstraintViolationMessage(name));
+        if (val == null) {
```

Comment:
I don't like this method.
We should validate value range the same way in SQL and in KV.
Can we check here only typeSpec equality (+ maybe nullability), and then pass `val` object to `NativeType.validate` method as `Column.type.validate(val);`
NativeType implementations could override the method to check bounds or length or whatever else.

Reply:
I'm ok to fix this in another ticket.

Reply:
I agree, this is very strange place, but it also not relates to initial issue and was approved by 2 reviewers ...
you can fill a different issue and it will be done there.